public/js/admin.js

```diff
@@ -474,34 +474,60 @@ var submitEditForm = function(id) {
     });
 }
 
+/* Updates the Vilibility of a Question */
 var updateVisibility = function(qid) {
     var question = {};
 
-    question['visible'] = $('#checked-' + qid).is(':checked');
-
-    $.ajax({
-        type: 'POST',
-        url: '/questionmod',
-        data: {
-            question: question,
-            id: qid
-        },
-        success: function(data) {
-            displayQuestionTable();
-            const msg = question['visible'] ? ' is now visible to the students' : ' is now&nbsp<u><b>not</b></u>&nbspvisible to the students';
-            const colour = question['visible'] ? colours.SUCCESS : colours.WARNING;
-            const warn = question['visible'] ? '<i class="material-icons">check</i>&nbsp&nbsp&nbsp' : '<i class="material-icons">warning</i>&nbsp&nbsp&nbsp';
-            dropSnack(colour, warn + 'Question ' + qid + msg);
+    swal({
+        type: 'warning',
+        title: 'Visibilty Change',
+        text: 'You are about to change the visibility status of question ' + qid,
+        showCancelButton: true,
+        showConfirmButton: true,
+        // User can only close the swal if they click one of the buttons
+        allowEscapeKey: false,
+        allowClickOutside: false,
+        // Clicking the buttons does not close the swal, it opens one of the swals below
+        closeOnConfirm: false,
+        closeOnCancel: false,
         },
-        error: function(data){
-            if (data['status'] === 401) {
-                window.location.href = '/';
+        function (isConfirm) {
+            // User confirms the visiblity change
+            if (isConfirm) {
+            swal('Changed!', 'The visibility of the quesiton has been changed', 'success');
```

Comment:
I don't think we should have that secondary SWAL. A Snackbar notif should be enough. (Less clicks are more convenient for the user.

Reply:
I agree when the user confirms the change, but what if they cancel. There is no snack bar notification, so I think I should keep the swal after the user cancels. What do you think?

Reply:
No need for anythjng when canceling.

Reply:
If you guys want you can make a swal that goes away after 2 seconds without the need for user input.

Reply:
The snack bar already notifies the admin that it was successful or not